Base/Python/slicer/util.py

```diff
@@ -2498,14 +2498,40 @@ def arrayFromTableColumnModified(tableNode, columnName):
 
 
 def updateTableFromArray(tableNode, narrays, columnNames=None):
-    """Set values in a table node from a numpy array.
-
-    :param columnNames: may contain a string or list of strings that will be used as column name(s).
-    :raises ValueError: in case of failure
-
-    Values are copied, therefore if the numpy array  is modified after calling this method,
-    values in the table node will not change.
-    All previous content of the table is deleted.
+    """Set values in a table node from a NumPy array or an array-like object (list/tuple of NumPy arrays).
+
+    This function can handle:
+
+      - **Structured (record) array** with named fields. Each field becomes a separate column.
```

Comment:
These are great as reference, but it would be really difficult to figure out what you mean exactly. Please add examples here and/or a `test_slicer_util_updateTableFromArray.py` test (that serve as examples) to demonstrate the ways this function can be used.

@jcfr's code snippet could be a good basis for a test, but instead of adding a new helper function `plotTableNode` we should test with `slicer.util.plot`.

Reply:
> re: test_slicer_util_updateTableFromArray

I updated it to check that structured array can be read and that column type and name are initialized accordingly.,

> re: we should test with `slicer.util.plot`
Good point.

Reply:
Tests have also been added